frame/curve-amm/src/lib.rs

```diff
@@ -227,13 +278,219 @@ pub mod pallet {
 
 			Ok(pool_id)
 		}
+
+		fn add_liquidity(
+			who: &Self::AccountId,
+			pool_id: PoolId,
+			amounts: Vec<Self::Balance>,
+			min_mint_amount: Self::Balance,
+		) -> Result<(), DispatchError> {
+			let zero = Self::Balance::zero();
+			ensure!(amounts.iter().all(|&x| x >= zero), Error::<T>::WrongAssetAmount);
+
+			let (provider, pool_id, token_amounts, invariant, token_supply, mint_amount) =
+				Pools::<T>::try_mutate(pool_id, |pool| -> Result<_, DispatchError> {
+					let pool = pool.as_mut().ok_or(Error::<T>::PoolNotFound)?;
+
+					let n_coins = pool.assets.len();
+
+					ensure!(n_coins == pool.balances.len(), Error::<T>::InconsistentStorage);
+
+					ensure!(n_coins == amounts.len(), Error::<T>::IndexOutOfRange);
+					let amp_f = FixedU128::saturating_from_integer(
+						u128::try_from(pool.amplification_coefficient)
+							.ok()
+							.ok_or(Error::<T>::Math)?,
+					);
```

Comment:
looks like it can rewrite so:
```
let amp_f = FixedU128::saturating_from_integer(
  u128::try_from(pool.amplification_coefficient).or(Error::<T>::Math)?,
);
```

Reply:
it seems to Vivek PR